src/tests/mock/http.rs

```diff
@@ -221,6 +227,43 @@ fn get(state: State, path: Option<String>, query: Option<String>) -> serde_json:
     assert_and_propagate_panics(assertions, &state.failed_assert)
 }
 
+/// Asserts that two JSON values are equal, excluding specified paths.
+///
+/// This macro is used to compare two JSON values for equality while ignoring
+/// certain paths in the JSON structure. The paths to be ignored are specified
+/// as a list of dot-separated strings. If the two JSON values are not equal
+/// (excluding the ignored paths), the macro will panic with a detailed error
+/// message indicating the location of the discrepancy.
+///
+/// # Arguments
+///
+/// * `$actual` - The actual JSON value obtained in a test.
+/// * `$expected` - The expected JSON value for comparison.
+/// * `$exclude_paths` - An array of dot-separated strings specifying the paths
+///   to be ignored during comparison.
+///
+/// # Panics
+///
+/// The macro panics if the actual and expected JSON values are not equal,
+/// excluding the ignored paths.
+///
+/// # Examples
+///
+/// ```
+/// let actual = json!({"user": {"id": 1, "name": "Alice", "email": "alice@example.com"}});
+/// let expected = json!({"user": {"id": 1, "name": "Alice", "email": "bob@example.com"}});
+/// assert_json_matches!(actual, expected, ["user.email"]);
+/// ```
+macro_rules! assert_json_matches {
+    ($actual:expr, $expected:expr, $exclude_paths:expr) => {{
+        // Encapsulate all operations inside a block to scope variables properly
+        let exclude_paths = parse_field_paths(&$exclude_paths);
+        let mut path = vec![];
```

Comment:
Defining an unused variable input just to make the function recursive doesn't seem to be the best practice. Why don't you make two functions? the wrapper (which defines this variable) and the one that calls itself?

Reply:
This macro is actually the wrapper.

Reply:
Added docs and created a helper function.